docs/swagger.yaml

```diff
@@ -2122,6 +2276,149 @@ paths:
           $ref: '#/responses/UnsupportedMediaType'
         '500':
           description: Unexpected internal errors.
+  /ldap/groups/search:
+    get:
+      summary: Search available ldap users.
+      description: >
+        This endpoint searches the available ldap users based on related
+        configuration parameters. Support searched by input ladp configuration,
+        load configuration from the system and specific filter.
+      parameters:
+        - name: groupname
+          in: query
+          type: string
+          required: false
+          description: Ldap group name
+      tags:
+        - Products
+      responses:
+        '200':
+          description: Search ldap group successfully.
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/LDAPGroups'
+        '500':
+          description: Unexpected internal errors.
+  /usergroups:
+   get:
+    summary: Get user group information
+    description: Get user group information
+    tags:
+     - Products
+    responses:
+      '200':
+        description: User group deleted successfully.
+      '400':
+        description: Illegal format of provided ID value.
+      '401':
+        description: User need to log in first.
+      '403':
+        description: User in session does not have permission to the user group.
+      '404':
+        description: user group does not exist.
+      '500':
+        description: Unexpected internal errors.      
+   post:
+      summary: Create user group
+      description: Create user group information
+      tags:
+        - Products    
+      parameters:
+        - name: usergroup
+          in: body
+          schema:
+            $ref: '#/definitions/UserGroup'
+      responses:
+        '200':
+          description: User group deleted successfully.
+        '400':
+          description: Illegal format of provided ID value.
+        '401':
+          description: User need to log in first.
+        '403':
+          description: User in session does not have permission to the user group.
+        '500':
+          description: Unexpected internal errors.        
+  /usergroups/{group_id}:
```

Comment:
If we treat group as a resource, it would be weird if we don't return group members in get.